statistics/histogram.go

```diff
@@ -232,20 +236,39 @@ func (c *Column) lowerBound(sc *variable.StatementContext, target types.Datum) (
 }
 
 // mergeBuckets is used to merge every two neighbor buckets.
-func (c *Column) mergeBuckets(bucketIdx int64) {
+func (hg *Histogram) mergeBuckets(bucketIdx int64) {
 	curBuck := 0
 	for i := int64(0); i+1 <= bucketIdx; i += 2 {
-		c.Buckets[curBuck] = bucket{
-			Count:   c.Buckets[i+1].Count,
-			Value:   c.Buckets[i+1].Value,
-			Repeats: c.Buckets[i+1].Repeats,
+		hg.Buckets[curBuck] = bucket{
+			Count:   hg.Buckets[i+1].Count,
+			Value:   hg.Buckets[i+1].Value,
+			Repeats: hg.Buckets[i+1].Repeats,
 		}
 		curBuck++
 	}
 	if bucketIdx%2 == 0 {
-		c.Buckets[curBuck] = c.Buckets[bucketIdx]
+		hg.Buckets[curBuck] = hg.Buckets[bucketIdx]
 		curBuck++
 	}
-	c.Buckets = c.Buckets[:curBuck]
+	hg.Buckets = hg.Buckets[:curBuck]
 	return
 }
+
+// Column represents a column histogram.
+type Column struct {
+	Histogram
+}
+
+func (c *Column) String() string {
+	return c.Histogram.toString(false)
+}
+
+// Index represents an Index histogram.
+type Index struct {
+	Histogram
+	NumColumns int
```

Comment:
I noticed that this value is setted somewhere, but not used. Does it have some future usage?